src/jobs/tracker_api.rs

```diff
@@ -1,30 +1,40 @@
 use std::sync::Arc;
 
 use log::info;
-use tokio::sync::oneshot::{self, Receiver};
+use tokio::sync::oneshot;
 use tokio::task::JoinHandle;
 
 use crate::api::server;
 use crate::tracker::TorrentTracker;
 use crate::Configuration;
 
 #[derive(Debug)]
-pub struct ApiReady();
+pub struct ApiServerJobStarted();
 
-pub fn start_job(config: &Configuration, tracker: Arc<TorrentTracker>) -> (JoinHandle<()>, Receiver<ApiReady>) {
+pub async fn start_job(config: &Configuration, tracker: Arc<TorrentTracker>) -> JoinHandle<()> {
     let bind_addr = config
         .http_api
         .bind_address
         .parse::<std::net::SocketAddr>()
         .expect("Tracker API bind_address invalid.");
 
-    let (tx, rx) = oneshot::channel::<ApiReady>();
-
     info!("Starting Torrust API server on: {}", bind_addr);
 
+    let (tx, rx) = oneshot::channel::<ApiServerJobStarted>();
+
+    // Run the API server
     let join_handle = tokio::spawn(async move {
-        server::start(bind_addr, tracker, tx).await;
+        if tx.send(ApiServerJobStarted()).is_err() {
+            panic!("the start job dropped");
+        }
+        server::start(bind_addr, tracker).await;
```

Comment:
What reason to send `ApiServerJobStarted()` before `server::start` has finished starting?

Maybe something like this is better:

```Rust
    let join_handle = tokio::spawn(async move {
        let handel = server::start(bind_addr, tracker);

        if tx.send(ApiServerJobStarted()).is_err() {
            panic!("the start job dropped");
        }

        handel.await
    });
```
?